app/policies/aal3_policy.rb

```diff
@@ -29,16 +33,35 @@ def aal3_configured_and_required?
     aal3_required? && @mfa_policy&.aal3_mfa_enabled?
   end
 
+  def piv_cac_only_setup_required?
+    piv_cac_only_required? && !piv_cac_enabled?
+  end
+
+  def piv_cac_only_required?
+    return if session.blank? || Figaro.env.allow_piv_cac_required != 'true'
+    sp_session = session.fetch(:sp, {})
+    return if sp_session.blank?
+    sp_session[:hspd12_piv_cac_requested]
```

Comment:
In [this comment](https://github.com/18F/identity-idp/pull/3945#discussion_r461089507) the changes I was referring to were that `session` and `sp_session` on this object went away, in favor of explicitly depending on properties in the initializer, so pushing `hspd12_piv_cac_requested` in to the calling in the controller

Reply:
Sorry but I'm not following.  we decided that we would not require them to send the piv_cac presented attribute to trigger piv/cac required.  how would we push hspd12_piv_cac_requested in to the controller?  we'd still be checking session.

Reply:
initially didn't see the init change on aal3_policy.  got it now.